lib/facter/easyrsa.rb

```diff
@@ -0,0 +1,46 @@
+Facter.add(:easyrsa) do
+  confine kernel: 'Linux'
+  setcode do
+    version = ''
+    binaryv2 = ''
+    binaryv3 = ''
+    operatingsystem = Facter.value(:operatingsystem)
+    operatingsystemrelease = Facter.value(:operatingsystemrelease)
+
+    case operatingsystem
+    when %r{RedHat|CentOS}
+      binaryv2 = '/usr/share/easy-rsa/2.0/pkitool'
+      binaryv3 = '/usr/share/easy-rsa/3/easyrsa'
+    when %r{Ubuntu|Debian}
+      case operatingsystemrelease
+      when %r{8|9|16.04|18.04}
+        binaryv2 = '/usr/share/easy-rsa/pkitool'
+        binaryv3 = '/usr/share/easy-rsa/easyrsa'
+      else
+        binaryv2 = '/usr/share/doc/openvpn/examples/easy-rsa/2.0/pkitool'
+        binaryv3 = '/usr/share/doc/openvpn/examples/easy-rsa/3.0/easyrsa'
+      end
+    when %r{Amazon}
+      binaryv2 = '/usr/share/easy-rsa/2.0/pkitool'
+      binaryv3 = '/usr/share/easy-rsa/3/easyrsa'
+    when %r{FreeBSD}
+      binaryv2 = '/usr/local/share/easy-rsa/pkitool'
+      binaryv3 = '/usr/local/share/easy-rsa/easyrsa'
+    end
+
+    if File.exist? binaryv2
```

Comment:
should we flip the logic here? v2 wil be used, even if v3 is available. Shouldn't we always use the latest available version?

Reply:
Ooooooh yes !